src/nix/flake.cc

```diff
@@ -783,8 +783,23 @@ struct CmdFlakeCheck : FlakeCommand
         }
 
         if (build && !drvPaths.empty()) {
-            Activity act(*logger, lvlInfo, actUnknown, fmt("running %d flake checks", drvPaths.size()));
-            store->buildPaths(drvPaths);
+            // This command doesn't need to actually substitute derivation outputs
+            // if they're missing but substitutable.
+            auto missing = store->queryMissing(drvPaths);
+            // Only occurs if `drvPaths` contains a `DerivedPath::Opaque`, which should never happen
+            assert(missing.unknown.empty());
+
+            std::vector<DerivedPath> toBuild;
+            for (auto & path : missing.willBuild) {
+                toBuild.emplace_back(
+                    DerivedPath::Built{
+                        .drvPath = makeConstantStorePathRef(path),
+                        .outputs = OutputsSpec::All{},
+                    });
+            }
+
+            Activity act(*logger, lvlInfo, actUnknown, fmt("running %d flake checks", toBuild.size()));
```

Comment:
This is another difference from the Determinate Nix implementation that I thought would be nice

It felt a bit confusing for Nix to report all checks were running when only a subset were actually being built. I think this makes the fact that this feature is even working *much* clearer, while also being more representative of what Nix is really doing

Alternatively, we could keep the total number of checks with a message like `fmt("running %d/%d flake checks", toBuild.size(), drvPaths.size())`